pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/AuthenticationRoute.java

```diff
@@ -1,101 +1,190 @@
+                            if (dsmToken != null) {
+                                return new DSMToken(dsmToken);
+                            }
+                            else {
+                                haltWithErrorMsg(401, response,  "DSMToken was null! Not authorized user");
+                            }
+                        }
+                        catch (DSMAuthenticationException e) {
+                            haltWithErrorMsg(401, response,  "DSMToken was null! Not authorized user", e);
+                        }
+                    }
+                    else {
+                        haltWithErrorMsg(400, response,  "user was null");
+                    }
+                }
+                catch (DSMAuthenticationException e) {
+                    haltWithErrorMsg(400, response,  "Problem getting user info from Auth0 token", e);
                 }
-                claims.put(authUserId, String.valueOf(userDto.getId()));
-                claims.put(authUserName, userDto.getName().orElse(""));
-                claims.put(authUserEmail, email);
+            }
+            else {
+                haltWithErrorMsg(400, response,  "There was no token in the payload");
+            }
+        }
+        catch (JsonSyntaxException e) {
+            haltWithErrorMsg(400, response,  "The provided JSON in the request was malformed", e);
+        }
+        return response;
+    }
+
+    private Map<String, String> getDSSClaimsFromOriginalToken(String auth0Token, String auth0Domain, Map<String, String> claims) {
+        Map<String, Claim> auth0Claims = Auth0Util.verifyAndParseAuth0TokenClaims(auth0Token, auth0Domain);
+
+        if(!auth0Claims.containsKey(tenantDomain) || !auth0Claims.containsKey(clientId) ||!auth0Claims.containsKey(userId)){
+            throw new RuntimeException("Missing dss claims in auth0 claims, can not authenticate");
+        }
+        claims.put(tenantDomain, auth0Claims.get(tenantDomain).asString());
+        claims.put(clientId, auth0Claims.get(clientId).asString());
+        claims.put(userId, auth0Claims.get(userId).asString());
+        //todo pegah get user id from database once DDP-7172 is done
 
-                long auth0Expiration = auth0UserInfo.getTokenExpiration();
-                int cookieAgeInSeconds = new Long(auth0Expiration - new Double(System.currentTimeMillis() / 1000d).intValue()).intValue();
+        return claims;
+    }
+
+    private String getNewAuth0TokenWithCustomClaims(Map<String, String> claims, String clientSecret, String clientId, String auth0Domain, String auth0Audience, String audienceNameSpace) throws DSMAuthenticationException {
+        String api = "/oauth/token";
+        String contentType = "application/x-www-form-urlencoded";
+        String clientCredentials = "client_credentials";
 
```

Comment:
can we move these strings into statics as well or is there a reason why we do them here?